cocotb/share/lib/simulator/simulatormodule.cpp

```diff
@@ -270,6 +270,24 @@ static PyObject *log_msg(PyObject *self, PyObject *args)
 }
 
 
+static p_callback_data make_callback_data(PyObject *func, PyObject *args, PyObject *kwargs)
+{
+    p_callback_data data = (p_callback_data)malloc(sizeof(s_callback_data));
+    if (data == NULL) {
+        PyErr_NoMemory();
+        return NULL;
```

Comment:
```suggestion
        return PyErr_NoMemory();
```

Reply:
Doesn't work, because `decltype(PyErr_NoMemory())` is `PyObject *` which is an incompatible pointer type.